cmd/registry/breaking-change-detector/breaking_change_detector.go

```diff
@@ -0,0 +1,237 @@
+			Modifications: make(map[string]*rpc.Diff_ValueChange),
+		},
+	}
+	updateChangeStatusAdditions(allChanges, d, diff)
+	updateChangeStatusDeletions(allChanges, d, diff)
+	updateChangeStatusModifications(allChanges, d, diff)
+
+	return allChanges
+}
+
+func updateChangeStatusAdditions(allChanges *rpc.ClassifiedChanges, d detectionTypeList, diff *rpc.Diff) {
+	additions := diff.GetAdditions()
+	for _, addition := range additions {
+		if fitsAnyPattern(d.breakingChanges.additions, addition) {
+			allChanges.BreakingChanges.Additions = append(allChanges.BreakingChanges.Additions, addition)
+			continue
+		}
+		if fitsAnyPattern(d.nonBreakingChanges.additions, addition) {
+			allChanges.NonBreakingChanges.Additions = append(allChanges.NonBreakingChanges.Additions, addition)
+			continue
+		}
+		allChanges.UnknownChanges.Additions = append(allChanges.UnknownChanges.Additions, addition)
+	}
+}
+
+func updateChangeStatusDeletions(allChanges *rpc.ClassifiedChanges, d detectionTypeList, diff *rpc.Diff) {
+	deletions := diff.GetDeletions()
+	for _, deletion := range deletions {
+		if fitsAnyPattern(d.breakingChanges.deletions, deletion) {
+			allChanges.BreakingChanges.Deletions = append(allChanges.BreakingChanges.Deletions, deletion)
+			continue
+		}
+		if fitsAnyPattern(d.nonBreakingChanges.deletions, deletion) {
+			allChanges.NonBreakingChanges.Deletions = append(allChanges.NonBreakingChanges.Deletions, deletion)
+			continue
+		}
+		allChanges.UnknownChanges.Deletions = append(allChanges.UnknownChanges.Deletions, deletion)
+	}
+}
+
+func updateChangeStatusModifications(allChanges *rpc.ClassifiedChanges, d detectionTypeList, diff *rpc.Diff) {
+	modifications := diff.GetModifications()
+	for modification, modValue := range modifications {
+		if fitsAnyPattern(d.breakingChanges.modifications, modification) {
+			allChanges.BreakingChanges.Modifications[modification] = modValue
+			continue
+		}
+		if fitsAnyPattern(d.nonBreakingChanges.modifications, modification) {
+			allChanges.NonBreakingChanges.Modifications[modification] = modValue
+			continue
+		}
+		if isNonStringValue(modValue.To) || isNonStringValue(modValue.From) {
+			if !isSameType(modValue.To, modValue.From) {
+				allChanges.BreakingChanges.Modifications[modification] = modValue
+				continue
+			}
+		}
+		allChanges.UnknownChanges.Modifications[modification] = modValue
+	}
+}
+
+func isSameType(valueOne, valueTwo string) bool {
+	if valueOne == valueTwo {
+		return true
+	}
+	if _, err := strconv.ParseInt(valueOne, 10, 64); err == nil {
+		if _, err := strconv.ParseInt(valueTwo, 10, 64); err == nil {
+			return true
+		}
+	}
+	if _, err := strconv.ParseFloat(valueOne, 64); err == nil {
+		if _, err := strconv.ParseFloat(valueTwo, 64); err == nil {
+			return true
+		}
+	}
+	if _, err := strconv.ParseBool(valueOne); err == nil {
+		if _, err := strconv.ParseBool(valueOne); err == nil {
+			return true
+		}
+	}
+	return false
+}
+
+func isNonStringValue(value string) bool {
```

Comment:
I think we've discussed this as a possible bug due to `rpc.Diff_ValueChange` only storing strings, regardless of the type used in the original spec file. How will this affect our ability to detect breaking changes?

We should document any caveats about this and write tests that target this detail. If the tests will fail due to the bug we can mark them as skipped, but they should still be checked into the suite.